### PR TITLE
adapted demo and migration tutorial to new package types

### PR DESCRIPTION
**Pull request by kaikreuzer**
See https://github.com/openhab/openhab-core/pull/81

Signed-off-by: Kai Kreuzer <kai@openhab.org>

### CODE REVIEW

File: tutorials/migration.md [hunk 1]
```diff
@@ -158,13 +158,12 @@ Next open for editing `addons.cfg`. This file contains parameters related to
 add-ons including allowing one to choose which add-ons to install. For now just
 set the following parameters:
 
-- `package = minimal` - minimal or standard are good choices here. Standard
-will include the standard UIs, though you can choose those individually later
-- `remote = true` - required for the installation of legacy 1.x and 
-experimental add-ons
+- `package = expert` - expert is good choices here. It will include the standard
```
Comment (BClark09) [inline tutorials/migration.md]: Expert is a good choice here

**Comment by kaikreuzer** [comment]
Thanks @BClark09, fixed it.

**Comment by ThomDietrich** [comment]
@rkoshak I was waiting on a comment from you regarding the migration tutorial. Lgtm.

**Comment by rkoshak** [comment]
Sorry, I've been struggling to keep up the last couple of weeks and didn't even see this.

**Comment by kaikreuzer** [comment]
@ThomDietrich & @rkoshak: I'd actually like to discuss the package options with you as I am not completely sure yet, whether these should be the final ones - I somehow feel that 5 are too many and 4 might be a better number.

But let me first briefly explain what this packaging at all does now: This setting is merely used at startup now to define an initial userdata/etc/org.openhab.addons.cfg file, if it does not exist (plus it sets simple linking mode for the "simple" package and it adds the demo files for the "demo" package). The addons file defines what should be installed for a start, but as Paper UI as well as conf/services/addons.cfg settings modify this, this is not "lasting", but the user is free to customize his setup in any way he wants.

Having to define this initial package in conf/services/addons.cfg before a first launch is at the moment only kept for backward compatibility. My plan would be to put the [package line in addons.cfg](https://github.com/openhab/openhab-distro/blob/master/distributions/distribution-resources/src/main/resources/conf/services/addons.cfg#L9) also in comments. If you do that for a test on a freshly unzipped distro, you will see a new intro page instead of the dashboard:

screen shot 2016-12-05 at 14 21 36

This page might explain you, why I prefer 4 instead of 5 options ;-) It is just an initial draft, it would still have to be polished with texts & images.
Clicking one of these options will simply set the package in the configuration, the very same way as it is done through the line in addons.cfg. What do you think of this mechanism? I think it is much more natural then having to go to a rather hidden file conf/services/addon.cfg before a first launch. My idea would also be to enhance this page in future to get hold of some other information, like the [geo location of the runtime](https://github.com/eclipse/smarthome/issues/1694) etc.
If you agree to pursue this way, we should probably also add dedicated pages on docs.openhab.org, which describe the package and its content - the setup page could then refer to these pages directly.

**Comment by rkoshak** [comment]
>  I somehow feel that 5 are too many and 4 might be a better number.

I agree. 5 seems too many. 4 seems right independent of how it looks on the page ;-). What isn't clear to me is what the difference is between Expert and Minimal. I would expect Expert to be just the core runtime also and I wouldn't expect anyone but an expert to use Minimal.

>  I think it is much more natural then having to go to a rather hidden file conf/services/addon.cfg before a first launch.

I agree. This is a great way to bootstrap the system for new users, particularly as we move more and more towards the UIs being fully capable for almost all users. And this can and will feed into a Getting Started tutorial.

In addition to getting geolocation, I can see this initial page being a great place to set just a few overall runtime settings like localization/language (inferred from geolocation perhaps?), and the initial user and password when authentication makes it back into OH. 

> If you agree to pursue this way, we should probably also add dedicated pages on docs.openhab.org, which describe the package and its content - the setup page could then refer to these pages directly.

Indeed and and these will also feature in many of the other tutorials so having the separate pages to point to makes a lot of sense.

**Comment by kaikreuzer** [comment]
>  I wouldn't expect anyone but an expert to use Minimal.

Well, "Expert" at the moment is the setup for OH1 users, i.e. it brings the compat layer and all the transformation services (which were part of the OH1 runtime and now became add-ons).
 "Minimal" just means that it does not install ANY add-on from the start and leaves it to the user. Right, this is something only experts will do. Maybe not worth to offer it as an option on the Setup page, but I wanted to provide a way to NOT start downloading add-ons that one in the end might not want to use at all for a custom setup.

**Comment by BClark09** [comment]
If they're looking at that page then they're already looking at a minimal setup right? Why not change the "Please choose your installation type:" to "Please choose your installation type, or to skip this process click [here](#):

Then offer the 4 other options?

**Comment by rkoshak** [comment]
> Well, "Expert" at the moment is the setup for OH1 users, i.e. it brings the compat layer and all the transformation services (which were part of the OH1 runtime and now became add-ons).

That makes a whole lot of sense. I forgot about the transformations.

Hmmmm, that pushes me back to five. Grrr. Having a configuration like this will make it much easier for OH 1.x migraters. 

I like @BClark09's suggestion. "Skip" becomes your minimal which can be a smaller button under the four options which will maintain the more even look while also giving the five options.

**Comment by kaikreuzer** [comment]
> I like @BClark09's suggestion. "Skip" becomes your minimal which can be a smaller button under the four options which will maintain the more even look while also giving the five options.

Ok, I like that as well. I can adapt the setup page accordingly.
But in the end it means we still have the 5 different packages, so the description in the addons.cfg in this PR stays unchanged, right?

**Comment by rkoshak** [comment]
> so the description in the addons.cfg in this PR stays unchanged, right?

I think so. Even if the GUI has the skip, the cfg file still needs to keep the minimal option.

**Comment by kaikreuzer** [comment]
Here's how it now looks like:

screen shot 2016-12-08 at 17 38 52

I'd like to add a link to the docs for some details on the packaging types. 
@ThomDietrich Any suggestion where this should point to? I can provide the content for such a page if you give me a space for it :-)

**Comment by rkoshak** [comment]
Nice! I like the background images. It really sets them apart. 

Does it make sense to put the docs under Installation? I could easily see these fitting in on the Overview page as it is something almost everyone will have to deal with no matter what approach they choose to install.

**Comment by ThomDietrich** [comment]
Nice! I like the idea, the design and the descriptions. The idea of @BClark09 comes out great in the new screenshot. This seems very intuitive and in line with how other systems would do this kind of thing :+1: 

One small things: "there are 4 pre-defined" -> "there are four pre-defined"?

Regarding the article: I would not put this inside "Installation". Installation addresses everything AROUND openHAB (OS, Hardware, Designer, ...). I would suggest to put the package details inside an article under Configuration ([screenshot](https://cloud.githubusercontent.com/assets/2870104/20943083/65305658-bbfe-11e6-9642-3076a16d01f0.png)) and (to make the link as stable as possible) as it's own document. How about `configuration/packages.md` which is linked as "Initial Setup" above the PaperUI entries in the screenshot?

**Comment by kaikreuzer** [comment]
> One small things: "there are 4 pre-defined" -> "there are four pre-defined"?

Addressed with https://github.com/openhab/openhab-core/pull/91

> How about configuration/packages.md which is linked as "Initial Setup" above the PaperUI entries (which are not final) in the screenshot?

Sounds good to me!

**Comment by ThomDietrich** [comment]
@kaikreuzer Done. https://github.com/openhab/openhab-docs/pull/148/commits/a64814b135dee2716f73063df5c3d4f89a0a62c0

How do you want to continue? Send me the content now or create your own PR after #148 is merged...

**Comment by kaikreuzer** [comment]
Won't be able to do it this weekend, so better after #148 has been merged.

**Pull request merged**